pam/integration-tests/helpers_test.go

```diff
@@ -308,7 +305,41 @@ func saveArtifactsForDebug(t *testing.T, artifacts ...string) {
 
 func maybeSaveFilesAsArtifactsOnCleanup(t *testing.T, artifacts ...string) {
 	t.Helper()
-	t.Cleanup(func() { saveArtifactsForDebug(t, artifacts...) })
+
+	t.Cleanup(func() {
+		if !t.Failed() && os.Getenv("AUTHD_TESTS_ARTIFACTS_ALWAYS_SAVE") == "" {
```

Comment:
I'm fine in not using the value stored if it may create troubles, but I'd still prefer to have this as a const, strings are fragile!

Reply:
I extracted a constant `alwaysSaveArtifactsEnvVar`